src/components/DeckMechanisms/index.js

```diff
@@ -158,10 +159,56 @@ export default class DeckMechanisms extends React.Component {
 
         // Remove the played card from the hand.
         newState.hand = state.hand.filter(cardId => cardId !== id)
-
+        
         if (options.discard) {
           return newState
         }
+        
+        // Turn one: Check if board is not full (by Rain of Frogs for example)
+        if (newState.turn === 1) {
+          const cardData = resolveCardForLevel(card)
```

Comment:
It seems like you do not actually need to resolve it since all you use is the level.